models/image_classification_mobilenet/demo.cpp

```diff
@@ -0,0 +1,1126 @@
+
+int main(int argc, char** argv)
+{
+    CommandLineParser parser(argc, argv, keys);
+
+    parser.about("Use this script to run classification deep learning networks in opencv Zoo  using OpenCV.");
+    if (argc == 1 || parser.has("help"))
+    {
+        parser.printMessage();
+        return 0;
+    }
+
+    int rszWidth = parser.get<int>("initial_width");
+    int rszHeight = parser.get<int>("initial_height");
+    bool swapRB = parser.get<bool>("rgb");
+    bool crop = parser.get<bool>("crop");
+    String model = parser.get<String>("model");
+    int backendTargetid = parser.get<int>("backend");
+
+    if (model.empty())
+    {
+        CV_Error(Error::StsError, "Model file " + model + " not found");
+    }
+    vector<string> labels = getLabelImagenet1k();
+
+    Net net = readNet(model);
+    net.setPreferableBackend(backendTargetPairs[backendTargetid].first);
+    net.setPreferableTarget(backendTargetPairs[backendTargetid].second);
+    //! [Open a video file or an image file or a camera stream]
+    VideoCapture cap;
+    if (parser.has("input"))
+        cap.open(parser.get<String>("input"));
+    else
+        cap.open(0);
+    if (!cap.isOpened())
+        CV_Error(Error::StsError, "Cannot opend video or file");
+    Mat frame, blob;
+    static const std::string kWinName = model;
+    while (waitKey(1) < 0)
+    {
+        cap >> frame;
+        if (frame.empty())
+        {
+            cout << "Frame is empty" << endl;
+            waitKey();
+            break;
+        }
+
+        if (rszWidth != 0 && rszHeight != 0)
+        {
+            resize(frame, frame, Size(rszWidth, rszHeight));
+        }
+        Image2BlobParams paramMobilenet;
+        paramMobilenet.datalayout = DNN_LAYOUT_NCHW;
+        paramMobilenet.ddepth = CV_32F;
+        paramMobilenet.mean = Scalar(123.675, 116.28, 103.53);
+        paramMobilenet.scalefactor = Scalar(1 / (255. * 0.229), 1 / (255. * 0.224), 1 / (255. * 0.225));
+        paramMobilenet.size = Size(224, 224);
+        paramMobilenet.swapRB = swapRB;
+        if (crop)
+            paramMobilenet.paddingmode = DNN_PMODE_CROP_CENTER;
+        else
+            paramMobilenet.paddingmode = DNN_PMODE_NULL;
+        //! [Create a 4D blob from a frame]
+        blobFromImageWithParams(frame, blob, paramMobilenet);
+
+        //! [Set input blob]
+        net.setInput(blob);
+        Mat prob = net.forward();
+
+        //! [Get a class with a highest score]
+        Point classIdPoint;
+        double confidence;
+        minMaxLoc(prob.reshape(1, 1), 0, &confidence, 0, &classIdPoint);
+        int classId = classIdPoint.x;
+        std::string label = format("%s: %.4f", (labels.empty() ? format("Class #%d", classId).c_str() :
+            labels[classId].c_str()),
+            confidence);
+        putText(frame, label, Point(0, 55), FONT_HERSHEY_SIMPLEX, 0.5, Scalar(0, 255, 0));
+        imshow(kWinName, frame);
+    }
+    return 0;
+}
+
+
+vector<string> getLabelImagenet1k()
```

Comment:
Can the classes names be out of the scope of sample?

https://github.com/opencv/opencv/blob/e60a7c0d490b5bd94524680deb74177d7bfb9ae4/samples/data/dnn/classification_classes_ILSVRC2012.txt#L697

Reply:
I think we just ship these labels along with the demo to make it work with less dependencies. For Python users, I would say very few of them have opencv source code. For C++ users, if they do not clone and build from source, they would not have opencv source code either.

Reply:
Yes I can do it but in [previous review](https://github.com/opencv/opencv_zoo/pull/171#discussion_r1225326383) I put label name in source file.
Reviewer must choose

Reply:
Thanks! This way, can we move them in a separate header and just `#include`?

Reply:
@dkurt I have got a problem translating python to c++ code for [text_recognition_rcnn demo](https://github.com/opencv/opencv_zoo/blob/main/models/text_recognition_crnn/crnn.py#L85)
In python code net.forward output is a numpy array : shape is known using outblob.shape

in C++ how can i guess blob size?

[here I put 24](https://github.com/LaurentBerger/opencv_zoo/blob/text_recognition_crnn/models/text_recognition_crnn/demo.cpp#L160) because I ran python code. It is for model text_recognition_CRNN_EN*.onnx
for model text_recognition_CRNN_CN_2021 and text_recognition_CRNN_CH I must use 25. Is it possible to guess those values without check model name?

Reply:
@LaurentBerger, you may access all dimensions by `size` array: `outputBlob.size[i]`. Opositting to `outputBlob.size()` it's multidimensional.